src/utilities.jl

```diff
@@ -16,10 +16,22 @@ function Base.randn(rng::AbstractVector{<:AbstractRNG}, T, dim::Int, n_chains::I
     return cat(randn.(rng, T, dim)...; dims=2)
 end
 
-# Sample from Categorical distributions
+# Helper functions to sync RNGs that ensures coupling works properly and reproducible
 
-randcat_logp(rng::AbstractRNG, unnorm_ℓp::AbstractVector) =
-    randcat(rng, exp.(unnorm_ℓp .- logsumexp(unnorm_ℓp)))
+rand_sync(rng::AbstractRNG, args...) = rand(rng, args...)
+
+# This function is needed to use only one RNG while a vector of RNGs are provided
+# in a way that RNGs are still synchronised. This is important if we want to use 
+# same RNGs to couple multiple chains.
+function rand_sync(rngs::AbstractVector{<:AbstractRNG}, args...)
```

Comment:
Consider renaming `rand` into `rand_coupled` for clarity?  
```suggestion
function rand_coupled(rngs::AbstractVector{<:AbstractRNG}, args...)
```

Reply:
I will do this on my local and push because there are other places we need to make this change.